src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java

```diff
@@ -1,52 +1,94 @@
-
-  protected static final GenericContainer<?> SURREAL_DB_CONTAINER =
-      new GenericContainer<>(DockerImageName.parse("surrealdb/surrealdb:latest"))
+  @Container
+  static final GenericContainer<?> surrealDbContainer =
+      new GenericContainer<>("surrealdb/surrealdb:latest")
           .withExposedPorts(8000)
-          .withCommand("start --user root --pass root");
-
-  @DynamicPropertySource
-  static void registerSurrealDbProperties(final DynamicPropertyRegistry registry) {
-    SURREAL_DB_CONTAINER.start();
-    String host = SURREAL_DB_CONTAINER.getHost();
-    Integer port = SURREAL_DB_CONTAINER.getMappedPort(8000);
-    registry.add("surrealdb.host", () -> host);
-    registry.add("surrealdb.port", port::toString);
-    registry.add("surrealdb.username", () -> "root");
-    registry.add("surrealdb.password", () -> "root");
-    registry.add("surrealdb.namespace", () -> "test_namespace");
-    registry.add("surrealdb.database", () -> "test_db");
-  }
+          .withCommand("start", "--log", "debug", "--user", "root", "--pass", "password")
+          .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("surrealDbContainer")));
+
+  private static final String TABLE = "page";
+  private static SurrealConnection connection;
+  private static SyncSurrealDriver driver;
+  private SurrealDbPageRepository repository;
 
   @BeforeAll
-  static void setUp() {
-    SURREAL_DB_CONTAINER.start();
+  static void setUpContainer() {
+    surrealDbContainer.start();
+
+    // Initialize SyncSurrealDriver with container connection
+    String host = surrealDbContainer.getHost();
+    Integer port = surrealDbContainer.getFirstMappedPort();
+
+    connection = new SurrealWebSocketConnection(host, port, false);
+    connection.connect(120); // timeout second
+
+    driver = new SyncSurrealDriver(connection);
+    driver.signIn("root", "password");
+    driver.use("test", "test");
   }
 
   @AfterAll
-  static void tearDown() {
-    SURREAL_DB_CONTAINER.stop();
+  static void tearDownContainer() {
+    connection.disconnect();
+    surrealDbContainer.stop();
+  }
+
+  @BeforeEach
+  void setUp() {
+    repository = new SurrealDbPageRepository(driver, FooterPage.class);
+  }
+
+  @AfterEach
+  void tearDown() {
+    driver.delete(TABLE);
   }
 
   @Test
-  @DisplayName("Should create and retrieve a ResourceContent entity")
-  void testSurrealDbConnection() {
-    assertTrue(SURREAL_DB_CONTAINER.isCreated());
-    assertTrue(SURREAL_DB_CONTAINER.isRunning());
+  void testSaveAndFindAll() {
+    // Arrange
+    List<Network> networks = createNetworksTest();
+    LabelLink link = createLinkTest();
+    FooterPage page1 =
+        new FooterPage(
+            "id", "footer_title", "footer_subtitle", "footer_description", networks, link);
+
+    FooterPage page2 =
+        new FooterPage(
+            "id1", "footer1_title", "footer1_subtitle", "footer1_description", networks, link);
+
+    // Act
+    repository.save(page1);
```

Comment:
```suggestion
    repository.create(page1);
```


I think that was some conflicts after I merged my changes, could you please have a look, probably change save to create, if you run locally I think you will see the exception quickly :) 

But besides that everything looks great! Amazing work